### DISCUSSION TITLE
Podcini is launching on Google Play and needs closed testing

### DISCUSSION THREAD

**Discussion by XilinJia** [XilinJia/Podcini#120] [Announcements]
> Please support the launch by participating the closed testing.  
> 
> You will first need to join this Google group: https://groups.google.com/g/podcini  
> 
> Then the testing can be accessed:
> 
> from Android: https://play.google.com/store/apps/details?id=ac.mdiq.podcini.R
> 
> or from the web: https://play.google.com/apps/testing/ac.mdiq.podcini.R
> 
> Signing key on Google Play is now the same as with Podcini apks elsewhere, so you can install the app from Google Play over an existing install (either the free or the play app) and continue to use, , so it doesn't give you much interruption by joining the closed testing, which is required for the formal launch on Google Play.
> 

> Formal listing on Google Play is finally approved. Thanks to everyone for the kind support.
> 
> I have't received notice for everyone participating the testing. If you have done that, and would like to be added to contributing member's list, as promised, I'd be happy to add you in. Honestly let me know.
> 
> Going forward, there will be testing releases on the "Open testing" track, accessible to anyone opting in for earlier testing on Google Play.  The first open-testing track release is version 7.0.  The open-testing release will be transferred to production release after a short period.
> 

Well, after approving it, Google rejected it with some unclear reasons, but it sounds like having to do with accessing to youtube contents.  So this is likely never listed on Google Play unless it was due to other reasons that can be amended. 

But thanks anyway to all for the support in testing it.

**Comment by toofyspooder** [comment]
Happy to help. Would prefer not to post me email though. Can I send another way?

**Reply by XilinJia** [reply]
Thank you.  Well understood.  Would it be better to do it through a google group?

**Reply by toofyspooder** [reply]
Whatever works for you really. Just not comfortable with posting my email for all to see.

**Comment by XilinJia** [comment]
Violates some family policy, will need to revise. stay tuned.

**Comment by Thunderhawk2001** [comment]
Link for play store is not working, but searching for podcini brings up this app:
![Screenshot_20241126-223254](https://github.com/user-attachments/assets/fd9651e4-7c1f-47ff-afdc-476ea0c3336e)


Is that it? If so: can't import my current database. If I try to import , it says that the file(PodciniBackup-2024-11-26.realm) is not a database.

**Reply by XilinJia** [reply]
Thank you for your effort and message.

The app you got must be of version 4 that I tried to launch long time ago but abandoned.

For the current app (Podcini.R), Google complains of violation of some family policy, so I need to sort it out. Will update the status shortly.

**Comment by XilinJia** [comment]
* Previous violation of some family policy has been cleared.  Please proceed to the testing.

**Comment by XilinJia** [comment]
6.15.2.1 on Google Play now has the same signing key as Podcini here, so it can be installed over an existing install (except if you installed 6.15.2 from Google Play, in which case, you need to uninstall it and install the new update).

**Reply by Thunderhawk2001** [reply]
Hi.

Hate to be that guy, but sadly it's not working. Getting the same "Podcini.R" cannot be installed" error message, when trying to update GitHub play version 6.15.2 via the Google Play store to version 6.15.2.1.
![Screenshot_20241203-100029](https://github.com/user-attachments/assets/0fa160db-6f6c-4235-8bbb-468df21d2de3)

**Reply by XilinJia** [reply]
Hum, sorry about that. 

6.15.2.1 is clearly using a different key from 6.15.2 on Play: I needed to uninstall the existing 6.15.2 from Play to install 6.15.2.1.  Let me find out why it can't install on an existing install from Github.

**Comment by XilinJia** [comment]
So Google Play's signing is complicated: it requires 2 sets of signing keys: app signing key and upload signing key. While I have both ready, approval for resetting upload signing key may take a couple days.  So we will wait a bit for getting a new version with compatibly signed apps.

**Comment by XilinJia** [comment]
Podcini on Google Play now has the same signing key as Podcini elsewhere, so it can be installed over an existing install.

**Reply by Thunderhawk2001** [reply]
Can confirm. It's possible to upgrade from GitHub version to Google Play store version.

**Comment by krjones91** [comment]
Two comments on this post. If it's preferred to split them, happy to do so. 
I just saw the comment regarding the 4.x abandonment and found the 6.x app on Google Play. Regarding aesthetics: the menu on the version 4 of the app looks more refined than version 6 (smaller font on v4, makes it less cramped). I'm not a developer, so I have no idea what goes into tweaking stuff like this, but from an end user standpoint, I prefer how the text and menus are rendered on the v4 app. 

I also have a feature request (again, not a dev so I have no idea what it would take to implement), it would be nice to have an option to limit the number of simultaneous downloads, which I find to be helpful either on lower-end devices or slow connections.

**Reply by XilinJia** [reply]
> the menu on the version 4 of the app looks more refined than version 6

Which menu are you referring to? Can you describe or post a screenshot?

**Reply by krjones91** [reply]
Screenshots attached.
![v6 downloads menu](https://github.com/user-attachments/assets/9e45fae3-3a5a-423c-baf2-61a438b1ed86)
![v4 downloads menu](https://github.com/user-attachments/assets/969f6f4e-87de-4440-829b-4be3fba736f6)
![v6 main menu](https://github.com/user-attachments/assets/07583d8f-e593-46f2-bf11-6e8b6bcec92d)
![v4 main menu](https://github.com/user-attachments/assets/582f6908-5733-4d0d-81d1-eb3e2212ea74)

**Reply by XilinJia** [reply]
I see and agree with you.  The new fonts are chosen on the first step of migration to Jetpack Compose.  I plan to refine them (and also the color scheme) later when most parts are migrated (Video player remains a major one now.)

**Reply by XilinJia** [reply]
> have an option to limit the number of simultaneous downloads

I checked the setup on this.  It's using the default provided by Android (supposedly Android knows about the condition of the device and provides an optimal number, max at 4 I guess). This number can be made to be reset by the user, but I'm not sure if it actually can make it more optimal.  

Have you noticed any problems with this?

**Reply by krjones91** [reply]
I wouldn't call it a "problem" necessarily. I just have noticed that on my current data connection (can be occasionally intermittent and lower data rate) that downloading one or two files at a time is more reliable than 3-4. However, this is a non-issue with a consistent network connection.

**Comment by IcedQuinn** [comment]
What is a "pre-IPO" member? Is Podcini issuing financial securities?

**Reply by XilinJia** [reply]
Anything is possible.

**Comment by XilinJia** [comment]
Formal listing on Google Play is finally approved.  Thanks to everyone for the kind support.

I have't received notice for everyone participating the testing.  If you have done that, and would like to be added to contributing member's list, as promised, I'd be happy to add you in.  Honestly let me know.

**Reply by toofyspooder** [reply]
That'd be cool!

I have a couple more bugs I'll be submitting also. Just haven't had a chance to get them over to you.
